packages/teams-js/src/internal/constants.ts

```diff
@@ -110,48 +112,21 @@ export const scanBarCodeAPIMobileSupportVersion = '1.9.0';
 
 /**
  * @hidden
- * List of supported Host origins
- *
- * @internal
- * Limited to Microsoft-internal use
- */
-export const validOrigins = [
-  'teams.microsoft.com',
-  'teams.microsoft.us',
-  'gov.teams.microsoft.us',
-  'dod.teams.microsoft.us',
-  'int.teams.microsoft.com',
-  'teams.live.com',
-  'devspaces.skype.com',
-  'ssauth.skype.com',
-  'local.teams.live.com', // local development
-  'local.teams.live.com:8080', // local development
-  'local.teams.office.com', // local development
-  'local.teams.office.com:8080', // local development
-  'outlook.office.com',
-  'outlook-sdf.office.com',
-  'outlook.office365.com',
-  'outlook-sdf.office365.com',
-  'outlook.live.com',
-  'outlook-sdf.live.com',
-  '*.teams.microsoft.com',
-  '*.www.office.com',
-  'www.office.com',
-  'word.office.com',
-  'excel.office.com',
-  'powerpoint.office.com',
-  'www.officeppe.com',
-  '*.www.microsoft365.com',
-  'www.microsoft365.com',
-  'bing.com',
-  'edgeservices.bing.com',
-  'www.bing.com',
-  'www.staging-bing-int.com',
-  'teams.cloud.microsoft',
-  'outlook.cloud.microsoft',
-  'm365.cloud.microsoft',
-  'copilot.microsoft.com',
-];
+ * Fallback list of valid origins
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export const validOriginsFallback = validOriginsJSON.validOrigins;
+
+/**
+ * @hidden
+ * CDN endpoint of the list of valid origins
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export const validDomainsCdnEndpoint = 'https://res-sdf.cdn.office.net/teams-js/validDomains/json/validDomains.json';
```

Comment:
>validDomainsCdnEndpoint [](http://example.com/codeflow?start=13&length=23)

Can this be typed as a `URL` object instead of a string?

Reply:
Yup, I'll make that change now